packages/Frontend/src/containers/AuthenticateContainer.jsx

```diff
@@ -23,47 +22,86 @@ class AuthenticateContainer extends Component {
     super(props);
     this.state = {
       email: '',
-      error: '',
+      error: false,
       loading: false,
       password: '',
     };
   }
 
+  componentDidMount = () => {
+    this.setState({ loading: true }, () => {
+      auth().getRedirectResult().then((result) => {
+        this.setState({ loading: false });
+        console.log('success', result);
+      }, (error) => {
+        this.setState({ loading: false });
+        // The provider's account email, can be used in case of
+        // auth/account-exists-with-different-credential to fetch the providers
+        // linked to the email:
+        console.error(error);
+        console.error(error.message);
+        console.error(error.code);
+        // In case of auth/account-exists-with-different-credential error,
+        // you can fetch the providers using this:
+        if (error.code === 'auth/account-exists-with-different-credential') {
+          auth().fetchProvidersForEmail(error.email).then((providers) => {
+            console.log('other providers', providers);
+          // The returned 'providers' is a list of the available providers
+          // linked to the email address. Please refer to the guide for a more
+          // complete explanation on how to recover from this error.
+          });
+        }
+      });
+    });
+  };
+
   updateFormByKey = key => ({ target }) => {
     this.setState({ [key]: target.value });
   };
 
-  authenticate = () => {
+  authenticate = (provider) => {
+    if (provider === 'google') {
+      return this.authenticateWithProvider(new auth.GoogleAuthProvider());
+    }
+    if (provider === 'facebook') {
+      return this.authenticateWithProvider(new auth.FacebookAuthProvider());
+    }
     if (this.props.match.url === signInRoute) {
       return this.signIn();
     }
     return this.signUp();
   }
 
   signIn = () => {
-    const { email, password } = this.state;
-    this.setState({ error: '', loading: true }, () => {
-      Auth.signIn(email, password)
-        .then(() => Auth.currentAuthenticatedUser())
-        .then(user => this.props.login(user))
-        .catch(error => this.setState({ error }))
-        .finally(() => this.setState({ loading: false }));
+    this.setState({ loading: true, error: false }, () => {
+      auth().signInWithEmailAndPassword(this.state.email, this.state.password)
+        .then(() => this.setState({ loading: false }))
+        .catch(error => this.setState({ loading: false, error }));
     });
   }
 
-  signUp = () => {}
+  signUp = () => {
+    auth().createUserWithEmailAndPassword(this.state.email, this.state.password)
+      .catch(error => this.setState({ error }));
+  }
+
+  authenticateWithProvider = (provider) => {
+    console.log('in authenticateWithProvider');
+    auth().useDeviceLanguage();
```

Comment:
I found useDeviceLanguage in a guide here: https://firebase.google.com/docs/auth/web/google-signin#handle_the_sign_in_flow_with_the_firebase_sdk

It tells facebook and google that when we redirect to their pages, use the language that the user has set in their browser. We can change this in the future, so if someone does set the source of truth to use Spanish, when they're redirected to google/facebook it'll also be in Spanish. Pretty cool.